libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -1322,4 +1325,40 @@ public void checkName(Class<?> callerClass, FileStore that) {
     public void checkType(Class<?> callerClass, FileStore that) {
         policyManager.checkReadStoreAttributes(callerClass);
     }
+
+    @Override
+    public void checkPathToRealPath(Class<?> callerClass, Path that, LinkOption... options) {
+        if (EntitlementChecker.class.isAssignableFrom(callerClass)) {
+            return;
+        }
+
+        boolean followLinks = true;
+        for (LinkOption option : options) {
+            if (option == LinkOption.NOFOLLOW_LINKS) {
+                followLinks = false;
+            }
+        }
+        if (followLinks) {
```

Comment:
I wonder if we need to care about links here. If a link is followed, they are only getting a path. If they want to go and read or write that path, they'll need the appropriate permission. But toRealPath is really just resolving the path. The only reason (I think) that we care to protect it is that it infers whether the path exists. But it's the original path that we care about there.

Reply:
yes, I did this to respect the existence check on the link destination (which would require a read permission otherwise)... happy to skip that check, actually had planned to raise that part for discussion, especially as doing the check here has the risk of stackoverflows, hence the check on the callerClass

Reply:
changed, not following links anymore